src/main/java/com/ibanity/apis/client/builders/IbanityServiceBuilder.java

```diff
@@ -0,0 +1,129 @@
+package com.ibanity.apis.client.builders;
+
+import com.ibanity.apis.client.helpers.IbanityService;
+import com.ibanity.apis.client.holders.ApplicationCertificateHolder;
+import com.ibanity.apis.client.holders.SignatureCertificateHolder;
+
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class IbanityServiceBuilder implements
+        OptionalPropertiesBuilder,
+        IbanityApiEndpointBuilder,
+        ApplicationPrivateKeyBuilder,
+        ApplicationPassphraseBuilder,
+        ApplicationCertificateBuilder,
+        RequestSignaturePrivateKeyBuilder,
+        RequestSignaturePassphraseBuilder,
+        RequestSignatureCertificateBuilder,
+        RequestSignatureCertificateIdBuilder {
+
+    private String apiEndpoint;
+    private String sslProtocol = "TLS";
+
+    private Certificate caCertificate;
+
+    private X509Certificate publicKey;
+    private PrivateKey privateKey;
+    private String privateKeyPassphrase;
+
+    private X509Certificate signaturePublicKey;
+    private PrivateKey signaturePrivateKey;
+    private String signaturePrivateKeyPassphrase;
+    private String signatureCertificateId;
+
+    public static IbanityApiEndpointBuilder builder() {
+        return new IbanityServiceBuilder();
+    }
+
+    public IbanityService build() {
+        SignatureCertificateHolder signatureCertificate = null;
+        if (signaturePrivateKey != null) {
+            signatureCertificate = new SignatureCertificateHolder(signaturePublicKey, signaturePrivateKey, signaturePrivateKeyPassphrase, signatureCertificateId);
+        }
+
+        ApplicationCertificateHolder applicationCertificate = new ApplicationCertificateHolder(publicKey, privateKey, privateKeyPassphrase, sslProtocol);
+        return new IbanityService(apiEndpoint, caCertificate, applicationCertificate, signatureCertificate);
+    }
+
+    public OptionalPropertiesBuilder caCertificate(Certificate certificate) {
+        this.caCertificate = certificate;
+        return this;
+    }
+
+    @Override
+    public OptionalPropertiesBuilder sslProtocol(String sslProtocol) {
```

Comment:
It depends if we want to allow customers to configure the protocol.
Might be useful in case of update for backward compatibility.
@thibaultponcelet @loicvigneron  Is it a feature we want or is it fine to hard  code TLS?